app/models/locu_data.rb

```diff
@@ -0,0 +1,67 @@
+class LocuData
+  include HTTParty
+  attr_reader :data, :name, :location, :categories, :open_hours, :menus
+  attr_accessor :venue_queries
+
+  def initialize(query = {})
+    @name = query['name']
+    @location = query['location']
+    @categories = query['categories']
+    @open_hours = query['open_hours']
+    @menus = query['menus']
+    @data = nil
+    @venue_queries = { 'categories' => { 'str_id' => 'restaurants' } }
+  end
+
+  def build_venue_queries
+    venue_queries['name'] = name if name
+    build_menus_query
+    venue_queries['open_hours'] = open_hours if open_hours
+    venue_queries['categories']['str_id'] = categories if categories
+    build_location_query
+    unless venue_queries['location']
+      venue_queries['location'] = {}
+      venue_queries['location']['locality'] = IpApi.new.data['city']
+    end
+    venue_queries
+  end
+
+  def search_function
+    default_fields = [ "name", "description", "website_url", "menus",
+      "open_hours", "categories", "location", "contact", "delivery" ]
```

Comment:
Align the elements of an array literal if they span more than one line.<br>Space inside square brackets detected.